### PR TITLE
util/s3: improve patch number capacity in version comparison

### PR DESCRIPTION
**Pull request by algorandskiy**
## Summary

Local builds could produce packages with higher than build number than the comparator in `updater` can handle.
This affects both updater and nodecfg.

## Test Plan

Added unit test

### CODE REVIEW

**Review by algobarb — DISMISSED** [review]
Thank you!

File: cmd/updater/version_test.go [hunk 1]
```diff
@@ -29,10 +29,10 @@ func TestGetVersion(t *testing.T) {
...
 	t.Parallel()
 
-	testValidVersion(t, "algonode_update_0.1.0.log", uint64(0x00010000))
-	testValidVersion(t, "algo_update_0.1.0", uint64(0x00010000))
-	testValidVersion(t, "algo_update_65535.1.0", uint64(0xFFFF00010000))
-	testValidVersion(t, "algo_update_65535.65535.65535", uint64(0xFFFFFFFFFFFF))
+	testValidVersion(t, "algonode_update_0.1.0.log", uint64(0x01000000))
+	testValidVersion(t, "algo_update_0.1.0", uint64(0x01000000))
+	testValidVersion(t, "algo_update_65535.1.0", uint64(0x00FFFF000001000000))
+	testValidVersion(t, "algo_update_65535.65535.65535", uint64(0x00FFFF00FFFF00FFFF))
```
Comment (winder) [inline cmd/updater/version_test.go:35]: May as well add the test
```suggestion
	testValidVersion(t, "algo_update_65535.65535.65535", uint64(0x00FFFF00FFFF00FFFF))
	testValidVersion(t, "algo_update_16777216.16777216.16777216", uint64(0xFFFFFFFFFFFFFFFFFF))
```

File: util/s3/s3Helper.go [hunk 2]
```diff
@@ -247,7 +247,7 @@ func GetVersionFromName(name string) (version uint64, err error) {
 	var val uint64
 	for index, match := range submatchAll[0] {
 		if index > 0 {
-			version <<= 16
+			version <<= 24
```
Comment (winder) [inline util/s3/s3Helper.go:250]: Is there a backwards compatibility issue? Hopefully these patch numbers transient?

**Review by winder — APPROVED** [review]
LGTM (assuming these numbers are never written down)

File: cmd/updater/version_test.go (hunk 1 above)
Comment (algorandskiy) [reply cmd/updater/version_test.go:35]: this number exceeds max uint64
maybe allocate 24bits for the patch and 16+16 for major/minor?

File: util/s3/s3Helper.go (hunk 2 above)
Comment (algorandskiy) [reply util/s3/s3Helper.go:250]: thank you for pointing out, it is used in `GetVersionPartsFromVersion` to back format human version, fixed + tests

File: cmd/updater/version_test.go (hunk 1 above)
Comment (winder) [reply cmd/updater/version_test.go:35]: 16/16/24 seems like a good option.

Comment (algorandskiy) [reply cmd/updater/version_test.go]: added

**Review by algobarb — APPROVED** [review]

**Review by winder — APPROVED** [review]

**Pull request merged**